### PR TITLE
Update runner images and compiler version

### PR DESCRIPTION
**Pull request by Thoemi09**

### CODE REVIEW

**Review by Wentzell — APPROVED** [review]
Thank you @Thoemi09 for this improvement!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -82,23 +81,28 @@ jobs:
...
+    - name: Add clang CXXFLAGS
       if: ${{ contains(matrix.cxx, 'clang') }}
       run: |
-        echo "PATH=/usr/local/opt/llvm/bin:$PATH" >> $GITHUB_ENV
         echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV
 
+    - name: Add clang LDFLAGS for macos to link against brew's libc++
+      if: ${{ contains(matrix.os, 'macos') && contains(matrix.cxx, 'clang') }}
+      run: |
+        echo 'LDFLAGS="-L$(brew --prefix llvm)/lib/c++ -L$(brew --prefix llvm)/lib -lunwind"' >> $GITHUB_ENV
```
Comment (hmenke) [inline .github/workflows/build.yml:104]: Single quotes will suppress expansion, this resulting in `LDFLAGS` being set literally to `-L$(brew --prefix llvm)/lib/c++ -L$(brew --prefix llvm)/lib -lunwind`, resulting in [errors](https://github.com/hmenke/w2dynamics_interface/actions/runs/13858838401/job/38781921730#step:10:201) like
```
ld: warning: search path '$(brew --prefix llvm)/lib/c++ -L$(brew --prefix llvm)/lib -lunwind' not found
```

```suggestion
        echo "LDFLAGS=-L$(brew --prefix llvm)/lib/c++ -L$(brew --prefix llvm)/lib -lunwind" >> $GITHUB_ENV
```

Comment (hmenke) [reply .github/workflows/build.yml:104]: Turns out, linking to libunwind also [doesn't work](https://github.com/hmenke/w2dynamics_interface/actions/runs/13859514589/job/38784227486#step:9:50).
```
    ld: library 'unwind' not found
```

```suggestion
        echo "LDFLAGS=-L$(brew --prefix llvm)/lib/c++ -L$(brew --prefix llvm)/lib" >> $GITHUB_ENV
```

Comment (Thoemi09) [reply .github/workflows/build.yml:104]: Hi Henri,

Thanks, this is definitely not correct. I don't remember why I put the LDFLAGS there in the first place. It seems to be working just fine without it. Have you tried removing it altogether?

**Pull request merged**